src/main/java/com/meilisearch/sdk/SearchRequest.java

```diff
@@ -336,4 +337,31 @@ private SearchRequest(
     public void setQuery(String q) {
         setQ(q);
     }
+
+    /**
+     * Method that returns the JSON String of the SearchRequest
```

Comment:
@alallema this class has a poor DX, it is very hard to use this kind of initializer structure because it will be very very hard to create an instance of a class with that many attributes to suit every possible case.

So I encourage you to add this change as a new TODO item in the refactor: "Make the `SearchRequest` class a builder".

Reply:
We can use here: https://projectlombok.org/features/Builder